pkg/kubelet/status/state/state_checkpoint.go

```diff
@@ -42,46 +42,43 @@ func NewStateCheckpoint(stateDir, checkpointName string) (State, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize checkpoint manager for pod allocation tracking: %v", err)
 	}
+
+	praInfo, err := restoreState(checkpointManager, checkpointName)
+	if err != nil {
+		//lint:ignore ST1005 user-facing error message
+		return nil, fmt.Errorf("could not restore state from checkpoint: %w, please drain this node and delete pod allocation checkpoint file %q before restarting Kubelet",
+			err, path.Join(stateDir, checkpointName))
+	}
+
 	stateCheckpoint := &stateCheckpoint{
-		cache:             NewStateMemory(),
+		cache:             NewStateMemory(praInfo.AllocationEntries),
 		checkpointManager: checkpointManager,
 		checkpointName:    checkpointName,
 	}
 
-	if err := stateCheckpoint.restoreState(); err != nil {
-		//lint:ignore ST1005 user-facing error message
-		return nil, fmt.Errorf("could not restore state from checkpoint: %v, please drain this node and delete pod allocation checkpoint file %q before restarting Kubelet", err, path.Join(stateDir, checkpointName))
-	}
+	klog.V(2).InfoS("State checkpoint: restored pod resource allocation state from checkpoint")
```

Comment:
Move this to line 75 so it doesn't get logged when there isn't a preexisting checkpoint.